app/common/directives/category-selector.directive.js

```diff
@@ -0,0 +1,118 @@
+module.exports = CategorySelectorDirective;
+
+CategorySelectorDirective.$inject = [];
+
+function CategorySelectorDirective() {
+    return {
+        restrict: 'E',
+        scope: {
+            available: '=',
```

Comment:
I suspect we could move some of the category loading and parent/child matching code out of other directives and in here too.

Rather than sending full categories into this directive, pass in an (optional) array of ids. Then load all categories from the TagsEndpoint and filter then in this directive. Right now we do this in the post-editor and its kinda messy.

Maybe thats too big a refactor though? Up to you...

Reply:
Fixed, but it needs a second look now...